src/shared/installationVerification.ts

```diff
@@ -284,24 +267,36 @@ export class InstallationVerification implements Verifier {
   public async streamTagGz(): Promise<NpmMeta> {
     const logger = await this.getLogger();
     const npmMeta = await this.retrieveNpmMeta();
+    if (!npmMeta.tarballUrl) {
```

Comment:
seems safe to update the npm meta type to account for `tarball` to always be present in the response.
https://github.com/verdaccio/verdaccio/blob/d52dbadae84edb1c34905bf09c3e8f8bf708eed0/packages/core/types/src/manifest.ts#L74

Reply:
you're right, but 😢 we've got a public constructor on an exported class, which would become invalid if the tarballUrl is required.

```
public constructor(private module: string, private version: string = 'latest', private cliRoot?: string) {
    this.npmMeta = {
      moduleName: module,
    };
  }
```

This is kinda like what I showed in demos with classes--all the props are optional and the consumers has to know which methods need be called before which props/methods become valid.